src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,229 @@
+            .address(Address.builder()
+                .postcode(POSTCODE)
+                .county(COUNTY)
+                .build())
+            .build();
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant).hasSize(1);
+        assertThat(valueInApplicant.get(0).get("value").equals(Applicant.builder()
+            .party(ApplicantParty.builder()
+                .partyId(null)
+                .partyType(null)
+                .name(null)
+                .address(Address.builder()
+                    .addressLine1(null)
+                    .addressLine2(null)
+                    .addressLine3(null)
+                    .postcode(POSTCODE)
+                    .country(COUNTRY)
+                    .county(null)
+                    .build())
+                .emailAddress(EmailAddress.builder()
+                    .email(null)
+                    .build())
+                .telephoneNumber(TelephoneNumber.builder()
+                    .telephoneNumber(null)
+                    .build())
+                .mobileNumber(MobileNumber.builder()
+                    .telephoneNumber(null)
+                    .build())
+                .jobTitle(null)
+                .pbaNumber(null)
+                .build())
+            .leadApplicantIndicator(null)
+            .build()));
+    }
+
+    private Applicant newApplicantBuilder() {
+        Applicant newApplicant = Applicant.builder()
+            .party(ApplicantParty.builder()
+                .partyId(PARTYID)
+                .partyType(PARTYTYPE)
+                .name(APPLICANT)
+                .address(Address.builder()
+                    .addressLine1(ADDRESSLINE1)
+                    .addressLine2(ADDRESSLINE2)
+                    .addressLine3(ADDRESSLINE3)
+                    .postTown(POSTTOWN)
+                    .postcode(POSTCODE)
+                    .county(COUNTY)
+                    .country(COUNTRY)
+                    .build())
+                .emailAddress(EmailAddress.builder()
+                    .email(EMAIL)
+                    .build())
+                .telephoneNumber(TelephoneNumber.builder()
+                    .telephoneNumber(TELEPHONE)
+                    .contactDirection(PERSONTOCONTACT)
+                    .build())
+                .mobileNumber(MobileNumber.builder()
+                    .telephoneNumber(MOBILE)
+                    .build())
+                .jobTitle(JOBTITLE)
+                .pbaNumber(PBANUMBER)
+                .build())
+            .leadApplicantIndicator(LEADAPPLICANTINDICATOR)
+            .build();
+        return newApplicant;
+    }
+
+    private OldApplicant getOldApplicant() {
```

Comment:
Get method does not quite fit here for me. In my mind get gives you something that is already present but here creation process happens. Bit misleading for me.

Reply:
👍